### ISSUE TITLE
Leo - Add model brand icons besides each one in the menu

### ISSUE THREAD

**Issue by aguscruiz** [brave/brave-browser#40033]
### Description

Since each model and company are getting more and more distinct and powerful, we should make it clearer that our available models come from them so we can entice users.

For that, in the model I added each model icon besides the name.

https://www.figma.com/design/MNwi0uwXNhkAbPiJ86G4vm/%E2%9C%A8-Leo?node-id=1506-7991&t=oQ6LKOYpPj990Os9-1

### Steps to reproduce

-

### Actual result

--

### Expected result

<img width="638" alt="image" src="https://github.com/user-attachments/assets/d0ca319a-38a8-4fab-affb-15c3220f009e">

### Reproduces how often

Easily reproduced

### Desktop Brave version (brave://version info)

--

### Android device

--

### Channel information

- [ ] release (stable)
- [ ] beta
- [X] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by fallaciousreasoning** [comment]
@aguscruiz we probably won't have a nice icon for local models so is there a good icon to display as a fallback?

**Comment by aguscruiz** [comment]
Yeah good point. Let's use `product-brave-leo` for those, until we can figure out if there's a way to capture the right ones. 

<img width="786" alt="Image" src="https://github.com/user-attachments/assets/95cee930-e212-4641-9c42-f65cffdd109d" />

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 16 BP31.250610.004 release-keys
Brave build: 1.82.139
Chromium: 139.0.7258.66 (Official Build) beta (64-bit)
```
---------------------------

**STEPS:**
1. Launch Brave
2. Open `Leo`
3. Tap the `Hamburger menu` in Leo's UI
4. *Verify*

**ACTUAL RESULTS:**
- Verified that LLM brand icons are shown for each model.
- Verified the same in `Dark` and `Light` modes.

---------------------------

| 1 | 2 | 3 | 4 | 5 |
|:-:|:-:|:-:|:-:|:-:|
| ![1](https://github.com/user-attachments/assets/5c2938a1-dfad-4653-9cbe-a106f711793a) | ![2](https://github.com/user-attachments/assets/0ddc6401-8e35-4af8-ac7b-a4ed405b6a3e) | ![3](https://github.com/user-attachments/assets/3a433b32-98fb-45e2-985f-309a7df57b11) | ![4](https://github.com/user-attachments/assets/a7d6719d-908b-4aea-b148-99417ac447bb) | ![5](https://github.com/user-attachments/assets/75181d3e-c5e8-4185-a54e-5f06e8c5aa4a) |
| ![1](https://github.com/user-attachments/assets/35134fe4-d740-4f02-9ca4-558f4c8b0054) | ![2](https://github.com/user-attachments/assets/0920a3b7-680b-4dc0-9657-a2d4ce04f637) | ![3](https://github.com/user-attachments/assets/1f81299a-ae97-4cf8-ae2c-03f8829705b6) | ![4](https://github.com/user-attachments/assets/36b85311-83b8-476a-a907-118c298f29b2) | ![5](https://github.com/user-attachments/assets/26dd2a7b-a319-4509-8bc6-495a05423913) |

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.82.138 Chromium: 139.0.7258.66 (Official Build) beta (64-bit)
-- | --
Revision | 7d0473c9495dcfc4fd5718c4b352298daccad654
OS | Windows 11 Version 24H2 (Build 26100.4652)
```

1. Installed 1.82.138
2. launched Brave
3. Opened Leo via sidebar and new tab
4. Verified each LLM model brand icons shown besides each one in the Language model menu
5. Confirmed the icons match to the design as per the Expected result in  https://github.com/brave/brave-browser/issues/40033#issue-2432197929 in light and dark themes

**_Sidepanel_**

Light | Dark
----- | ----
<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/4cb22bc6-4249-4bae-898a-a14bcf3d1a02" />|<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/aeb15b06-2cbc-4570-9cfd-0aa2c9628559" />
<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/cccedda2-3225-4c60-9ae2-2e307759b4bf" />|<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/1f97808b-efe3-4298-9f97-65386dbb815d" />
<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/c0420f6a-8491-467d-a812-c9a0d874a836" />|<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/77560003-dd32-43a3-9627-bf7fc808a483" />
<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/2bca1fa1-d96e-4d47-9c32-59de05d3fcff" />|<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/2f2d9e00-d50c-4690-949d-aa458e2cead4" />
<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/3dcde5f5-f2ec-44c5-9d2d-5e3956e9f045" />|<img width="399" height="932" alt="Image" src="https://github.com/user-attachments/assets/9b50afb2-f1f4-48f6-ab16-772a53dc4ab4" />

**_New tab/full page chat_**

Light | Dark
----- | ----
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/d6c0e70f-adff-4d8b-a98f-722a1b1cc382" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/4d5dab90-0130-4c20-b373-af7314f1d8a2" />
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/a9ce5cd4-2077-466f-a71a-1cb127a9e2a1" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/0c0063e0-06c7-43b5-91b7-cf7980c3880f" />
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/945da184-c716-46bb-9bea-c7eead3100e0" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/42a0b7dc-1b05-4899-bd4b-920eecab5aa4" />
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/4efb7b09-a26e-4306-a918-d60a80047120" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/af1eb522-f8a7-4029-9868-ae87c30aed63" />
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/30a7203a-426c-416c-b6f9-48b0874fe345" />|<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/acc2f5d3-de91-4cab-bfe2-244140d7114a" />
